model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/UserAdapter.java

```diff
@@ -174,20 +175,39 @@ public void setEnabled(boolean enabled) {
 
     @Override
     public void setSingleAttribute(String name, String value) {
-        getDelegateForUpdate();
         if (UserModel.USERNAME.equals(name) || UserModel.EMAIL.equals(name)) {
             value = KeycloakModelUtils.toLowerCaseSafe(value);
         }
+        if (updated == null) {
+            Set<String> oldEntries = getAttributeStream(name).collect(Collectors.toSet());
+            Set<String> newEntries = value != null ? Set.of(value) : Collections.emptySet();
+            if (Objects.equals(oldEntries, newEntries)) {
```

Comment:
Can you please use `org.keycloak.common.util.CollectionUtil#collectionEquals`. It is the algorithm we are using to avoid unnecessary updates when the values are the same. It does not take into account the order.